server/broadcast.go

```diff
@@ -389,21 +388,30 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 			cond.L.Unlock()
 		}()
 
-		if bos := sess.BroadcasterOS; bos != nil && !drivers.IsOwnExternal(url) {
-			data, err := drivers.GetSegmentData(url)
+		bos := sess.BroadcasterOS
+
+		var data []byte
+		// Download segment data in the following cases:
+		// - A verification policy is set. The segment data is needed for signature verification and/or pixel count verification
+		// - The segment data needs to be uploaded to the broadcaster's own OS
+		if verifier != nil || (bos != nil && !drivers.IsOwnExternal(url)) {
```

Comment:
NB. The broadcaster will unnecessarily download segment data in the following scenario:

- Off-chain mode
- Using a verification policy with a verifier that counts pixels on its own
- Using an external OS

Technically, the broadcaster does not need to download segment data here in that case. I chose not to handle that case right now as it requires the broadcaster to know if the policy's verifier will count pixels on its own.

Reply:
Does unnecessarily downloading the segment data impact the latency for the off-chain workflow in scope of the video user team and pilots we are doing? 

If it does it might be worthwhile to address but I have a hard time judging this from my perspective.

Reply:
> Does unnecessarily downloading the segment data impact the latency for the off-chain workflow in scope of the video user team and pilots we are doing?

The unnecessary segment download should not impact the off-chain workflow because the off-chain workflow typically does not use a verification policy. The only time an off-chain B would use a verification policy is for internal verification specific testing. I'm inclined to leave this as-is for now since the inefficiency is isolated to verification specific testing scenarios and because additional verification policy changes such as livepeer/verification-classifier#64 may affect the choice of solution for resolving this inefficiency.